src/EnergyPlus/UnitarySystem.cc

```diff
@@ -11933,7 +11934,90 @@ namespace UnitarySystems {
                 PartLoadFrac = 0.0;
                 CompOn = 0;
 
-                if (CoilType_Num == DataHVACGlobals::CoilDX_CoolingSingleSpeed) { // COIL:DX:COOLINGBYPASSFACTOREMPIRICAL
+                if (this->m_EMSOverrideCoilSpeedNumOn &&
+                    (CoilType_Num == DataHVACGlobals::CoilDX_MultiSpeedCooling || CoilType_Num == DataHVACGlobals::CoilDX_Cooling)) {
+                    this->m_CoolingSpeedNum = ceil(this->m_EMSOverrideCoilSpeedNumValue);
+                    this->m_SpeedNum = this->m_CoolingSpeedNum;
+                    bool useMaxedSpeed = false;
+                    if (this->m_SpeedNum > this->m_NumOfSpeedCooling) {
+                        this->m_CoolingSpeedNum = this->m_NumOfSpeedCooling;
+                        this->m_SpeedNum = this->m_NumOfSpeedCooling;
+                        CycRatio = this->m_CoolingCycRatio = 1.0;
+                        PartLoadFrac = SpeedRatio = this->m_CoolingSpeedRatio = 1.0;
+                        useMaxedSpeed = true;
+                        if (this->m_CoilSpeedErrIdx == 0) {
+                            ShowWarningMessage(state, "Wrong coil speed EMS override value, for unit=\"" + this->m_CoolingCoilName);
+                            ShowContinueError(state,
+                                              "  Exceeding maximum coil speed level. Speed level is set to the maximum coil speed level allowed.");
+                        }
+                        ShowRecurringWarningErrorAtEnd(
```

Comment:
This is better. I was going to comment that the two error messages here will conflict once the `m_CoilSpeedErrIdx` is set. But now I see that `ShowRecurringWarningErrorAtEnd` ignores the incoming index and searches for a matching error string. A test file which sets the coil speed to values higher than max peed, and to negative values, does produce both recurring error messages. So, this is fine as-is.